### PR TITLE
Cache run results

### PR DESCRIPTION
**Pull request by MasloMaslane**
I added caching test results. Also when checker is changed, all cached results are removed.

### CODE REVIEW

**Comment by MasloMaslane** [comment]
I added new function: when a file from `extra_compilation_files` changes, all solutions in the same language as the file are removed from cache

File: src/sinol_make/helpers/cache.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    """
+    os.makedirs(paths.get_cache_path("md5sums"), exist_ok=True)
+    try:
+        with open(paths.get_cache_path("md5sums", os.path.basename(solution_path)), 'r') as cache_file:
+            data = yaml.load(cache_file, Loader=yaml.FullLoader)
+            if not isinstance(data, dict):
+                return CacheFile()
+            return CacheFile.from_dict(data)
+    except (FileNotFoundError, yaml.YAMLError, TypeError):
+        return CacheFile()
```
Comment (tonowak) [inline src/sinol_make/helpers/cache.py:24]: Maybe we should make a warning about "corrupted" files?

File: src/sinol_make/helpers/cache.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    info.save(file_path)
+
+    if is_checker:
+        for solution in os.listdir(paths.get_cache_path('md5sums')):
+            info = get_cache_file(solution)
+            info.tests = {}
+            info.save(solution)
+
+
+def check_extra_compilation_files(extra_compilation_files, task_id):
```
Comment (tonowak) [inline src/sinol_make/helpers/cache.py:65]: Can you try to come up with a name that more clearly shows what this function does?

File: src/sinol_make/structs/status_structs.py [hunk 3]
```diff
@@ -15,3 +17,88 @@ def __str__(self):
...
+        elif status == "ML":
+            return Status.ML
+        elif status == "RE":
+            return Status.RE
+        elif status == "WA":
+            return Status.WA
+        elif status == "OK":
+            return Status.OK
+        else:
+            return Status.PENDING
```
Comment (tonowak) [inline src/sinol_make/structs/status_structs.py:36]: is there a `status` string that means `PENDING`? Maybe we should warn/error when the content of `status` doesn't make sense?

**Review by tonowak — APPROVED** [review]
Looks good! Some small suggestions, merge after resolving them:

**Pull request merged**